config/s2i/gold/master/configuration/init.groovy

```diff
@@ -0,0 +1,37 @@
+import java.util.logging.Logger
+import jenkins.security.s2m.*
+import jenkins.model.*;
+import hudson.markup.RawHtmlMarkupFormatter
+import hudson.model.*
+import hudson.security.*
+import com.redhat.jenkins.plugins.ci.*
+import com.redhat.jenkins.plugins.ci.messaging.*
+
+def logger = Logger.getLogger("")
+logger.info("Disabling CLI over remoting")
+jenkins.CLI.get().setEnabled(false);
+logger.info("Enable Slave -> Master Access Control")
+Jenkins.instance.injector.getInstance(AdminWhitelistRule.class).setMasterKillSwitch(false);
+// Set global and job read permissions
+def strategy = Jenkins.instance.getAuthorizationStrategy()
+strategy.add(hudson.model.Hudson.READ,'anonymous')
+strategy.add(hudson.model.Item.READ,'anonymous')
+Jenkins.instance.setAuthorizationStrategy(strategy)
+// Set Markup Formatter to Safe HTML so PR hyperlinks work
+Jenkins.instance.setMarkupFormatter(new RawHtmlMarkupFormatter(false))
+Jenkins.instance.save()
+
+logger.info("Setup fedora-fedmsg Messaging Provider")
```

Comment:
I don't see jms messaging in the plugins.txt. Unsure how this works without the plugin installed. Not sure if there is magic behind it and it will like add the plugin for you or not (not sarcastic, actually don't know what will happen here).

Reply:
I don't know either.  That init.groovy is either used in or a part of a larger init.groovy for each project, but not all projects have the jms-messaging plugin.  I don't see anything wrong with adding it.

Reply:
This init.groovy code adds a fedmsg message provider to the master. I am pretty sure it depends on jms-messaging being present. I would be surprised if there was a master with this code, but without jms-messaging as a plugin. Without jms-messaging, I expect this to do nothing or fail, just not sure which